qiskit_nature/second_q/circuit/library/initial_states/vscf.py

```diff
@@ -52,32 +53,141 @@ def __init__(
                              that of the :class:`DirectMapper`. However, for future-compatibility of
                              this functions signature, the argument has already been inserted.
         """
-        # get the bitstring encoding initial state
-        bitstr = vscf_bitstring(num_modals)
-
-        # encode the bitstring in a `VibrationalOp`
-        label = ["+" if bit else "I" for bit in bitstr]
-        bitstr_op = VibrationalOp("".join(label), num_modes=len(num_modals), num_modals=num_modals)
-
-        # map the `VibrationalOp` to a qubit operator
-        if qubit_converter is not None:
+        super().__init__()
+        self._num_modals = num_modals
+        self._qubit_converter = None
+        self._bitstr: list[bool] | None = None
+
+        self.qubit_converter = qubit_converter
+
+        if num_modals is not None:
+            self.num_modals = num_modals
```

Comment:
```suggestion
        self._num_modals = num_modals
        self._qubit_converter = None
        self._bitstr: list[bool] | None = None

        self.qubit_converter = qubit_converter
```
Here we cannot get rid of the `qubit_converter` register trigger since it has side effects, but the `num_modals` setting is not needed